packages/sdk/src/evm/core/classes/smart-wallet.ts

```diff
@@ -0,0 +1,461 @@
+
+      if(restrictionsForSigner.role === this.emptyRole) {
+        throw new Error("Signer does not have any access");
+      }
+
+      if(restrictionsForSigner.approvedTargets.includes(target)) {
+        throw new Error("Target is already approved");
+      }
+
+      const newTargets = [...restrictionsForSigner.approvedTargets, target];
+
+      const newRestrictions: IAccountPermissions.RoleRestrictionsStruct = { ...restrictionsForSigner, approvedTargets: newTargets };
+      return Transaction.fromContractWrapper({
+        contractWrapper: this.contractWrapper,
+        method: "setRoleRestrictions",
+        args: [newRestrictions],
+      })
+    }
+  )
+  
+  /**
+   * Disapprove an address as a call target for a given signer on the smart wallet.
+   *
+   * @remarks Disapprove an address as a call target for a given signer on the smart wallet.
+   * 
+   * @param signer - A signer with restricted access to the smart wallet.
+   * @param target - The address to disapprove as a call target for the signer.
+   * 
+   * @example
+   * ```javascript
+   * const tx = await contract.smartWallet.disapproveTargetForSigner(signer, target);
+   * const receipt = tx.receipt();
+   * ```
+   *
+   * @twfeature SmartWallet
+   */
+  disapproveTargetForSigner = buildTransactionFunction(
+    async(
+      signer: string,
+      target: string,
+    ): Promise<Transaction> => {
+      const restrictionsForSigner: IAccountPermissions.RoleRestrictionsStruct = await this.contractWrapper.readContract.getRoleRestrictionsForAccount(signer);
+
+      if(restrictionsForSigner.role === this.emptyRole) {
+        throw new Error("Signer does not have any access");
+      }
+
+      if(!restrictionsForSigner.approvedTargets.includes(target)) {
+        throw new Error("Target is not already approved");
+      }
+
+      const newTargets = restrictionsForSigner.approvedTargets.filter((approvedTarget) => ethers.utils.getAddress(approvedTarget) !== ethers.utils.getAddress(target));
+
+      const newRestrictions: IAccountPermissions.RoleRestrictionsStruct = { ...restrictionsForSigner, approvedTargets: newTargets };
+      return Transaction.fromContractWrapper({
+        contractWrapper: this.contractWrapper,
+        method: "setRoleRestrictions",
+        args: [newRestrictions],
+      })
+    }
+  )
+
+  /**
+   * Update an address' access to the smart wallet.
+   *
+   * @remarks Updates an address' access to the smart wallet.
+   * 
+   * @param signer - The address whose access to the smart wallet is to be updated.
+   * @param restrictions - The restrictions to be applied to the signer's use of the smart wallet.
+   * 
+   * @example
+   * ```javascript
+   * const tx = await contract.smartWallet.updateAccess(signer, restrictions);
+   * const receipt = tx.receipt();
+   * ```
+   *
+   * @twfeature SmartWallet
+   */
+  updateAccess = buildTransactionFunction(
+    async(
+      signer: string,
```

Comment:
i feel like we should call these inputs `signerAddress` to be explicit. Signer generally means the object